<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro">

    <xacro:if value="$(arg sim_mode)">
        <ros2_control name="GazeboSimSystem" type="system">
            <hardware>
                <!-- Gazebo Sim (gz_ros2_control) plugin for Ignition Fortress -->
                <plugin>gz_ros2_control/GazeboSimSystem</plugin>
            </hardware>
            <joint name="joint_1">
                <command_interface name="effort">
                    <param name="min">-1000</param>
                    <param name="max">1000</param>
                </command_interface>
                <state_interface name="velocity"/>
                <state_interface name="position"/>
                <state_interface name="effort"/>
            </joint>
            <joint name="joint_2">
                <command_interface name="effort">
                    <param name="min">-1000</param>
                    <param name="max">1000</param>
                </command_interface>
                <state_interface name="velocity"/>
                <state_interface name="position"/>
                <state_interface name="effort"/>
            </joint>
            <joint name="joint_3">
                <command_interface name="effort">
                    <param name="min">-1000</param>
                    <param name="max">1000</param>
                </command_interface>
                <state_interface name="velocity"/>
                <state_interface name="position"/>
                <state_interface name="effort"/>
            </joint>
        </ros2_control>
    </xacro:if>

    <!-- Gazebo Sim (Ignition Fortress) ros2_control plugin -->
    <gazebo>
        <plugin filename="libgz_ros2_control-system.so" name="gz_ros2_control::GazeboSimROS2ControlPlugin">
            <parameters>$(find arm_bot)/config/my_controllers.yaml</parameters>
            <parameters>$(find arm_bot)/config/gaz_ros2_ctl_use_sim.yaml</parameters>
        </plugin>
    </gazebo>

</robot>
